node/src/components/consensus/highway_core/state.rs

```diff
@@ -245,8 +293,12 @@ impl<C: Context> State<C> {
 
     pub(crate) fn add_evidence(&mut self, evidence: Evidence<C>) {
         let idx = evidence.perpetrator();
+        if Some(&Fault::Banned) == self.faults.get(&idx) {
```

Comment:
It _could_ be `Fault::Direct` or `Fault::Indirect` as well, right? For those cases we would also return early.

Reply:
You're right, for `Direct` we might as well return early. It basically doesn't matter: The new evidence is as good as the old one.

For `Indirect`, however, we actually do want to replace our entry with the new `Direct` evidence.